### ISSUE TITLE
Accept a wider range of versions for dependencies

### ISSUE THREAD

**Issue by FelixSchwarz** [quic/aimet#2484]
Currently when building an aimet wheel there are very strict requirements for dependency versions. I understand that this is likely necessary for the "big ticket items" like pytorch (#2483) and tensorflow but do you really need `Jinja2==3.0.3` and no other version?

The current approach makes it really hard to use aimet wheels outside of docker. It would be easier for us to integrate the library if it had less strict requirements and loosening the versions for the "simple" dependencies might be a first step.

Please note that this is not about supporting any version. You could decide on a sensible set of versions for your CI but it would be helpful if users you run the library with different versions more easily.

**Comment by quic-mangal** [comment]
@FelixSchwarz, yes definitely. At times we need to fix these versions because of test failures. We haven't reviewed them in a while but I will create an AI to review these. Thanks for the suggestion.

**Comment by FelixSchwarz** [comment]
Thank you for considering the request.

> At times we need to fix these versions because of test failures.

Just a suggestion: In my projects I always add a comment when setting any kind of version restriction. Sometimes the reason is just "we don't bother testing the ancient foo 1.x but only 2.x which contains major API changes" but it can also be more specific like ">= 1.2.3 due to issue 1234 <github url>, seeing in CI running with Ubuntu 22.04".

**Comment by FelixSchwarz** [comment]
Additional data point: Today I learned that another group (>> 100 developers) uses aimet but decided to just to strip all dependency information from the wheels, basically manually copying the "aimet_..." directories to the target because aimet's dependencies were too strict and did not match their environment.

**Comment by quic-mangal** [comment]
Yes, definitely agreed some requirements might be outdated.

**Comment by quic-kyunggeu** [comment]
This issue is resolved in aimet >= 2.0. We unpinned all unnecessary version pinning
